code/modules/projectiles/special.dm

```diff
@@ -674,6 +674,93 @@ ABSTRACT_TYPE(/datum/projectile/special)
 					boutput(dropme, __red("Your coffin was lost or destroyed! Oh no!!!"))
 		..()
 
+/datum/projectile/special/homing/magicmissile
+	name = "magic missile"
+	sname = "magic missile"
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "magicm"
+	shot_sound = null
+	power = 10
+	cost = 1
+	damage_type = D_KINETIC
+	dissipation_delay = 0
+	dissipation_rate = 0
+	ks_ratio = 1
+	brightness = 2
+	projectile_speed = 10
+	is_magical = 1 // It passes right through them, but just for consistency
+	auto_find_targets = 0
+	min_speed = 0.1
+	max_speed = 10
+	goes_through_walls = 0 // It'll stop homing when it hits something, then go bouncy
+	var/max_bounce_count = 3 // putting the I in ICEE BEEYEM
+	var/weaken_length = 5 SECONDS
+	var/impact_brute = 5
+	var/bouncy = 0 // A fallback for when a missile is shot without a target
+	var/slam_text = "The magic missile SLAMS into you!"
+	var/hit_sound = 'sound/effects/mag_magmisimpact_bounce.ogg'
+	var/cat_sound = 'sound/voice/animal/cat.ogg'
+	var/last_sound_time = 0
+	var/last_person_hit // Dont spam the fact you didn't hit them, its rude
+
+	on_pre_hit(var/atom/hit, var/angle, var/obj/projectile/O)
+		. = ..()
+		if(isliving(hit))
+			var/mob/living/M = hit
+			if (iswizard(M) || M.traitHolder?.hasTrait("training_chaplain"))
+				if(M?.name != src.last_person_hit)
+					boutput(M, "The magic missile passes right through you!")
+				last_person_hit = M?.name
+				return TRUE
+			else
+				return FALSE
+		else if(src.homing_active && isturf(hit) && hit?.density)
+			src.homing_active = 0
+			src.bouncy = 1
+		/// Missiles home into their targets until they hit a wall, forget their target, and then just bounce around
+		else if(!src.homing_active)
+			if(isturf(hit))
+				return FALSE
+			else
+				return TRUE
```

Comment:
I don't really like this proc, can't put my finger on why though.

Reply:
Don't worry, there are plenty of reasons not to like it. Such as that it uses a hacky proc that early-returns Collide() if it returns anything. Or that it only checks for turfs for whether it should stop homing. Or that it gets called every frame that it happens to intersect with something. Or that each frame it's inside something it didnt collide with, it does two typechecks.

holy shit this is an awful proc

Reply:
made awful proc less awful, it actually works now!